pandas/_libs/algos.pyx

```diff
@@ -674,12 +677,15 @@ def backfill_2d_inplace(algos_t[:, :] values,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def is_monotonic(ndarray[algos_t, ndim=1] arr, bint timelike):
+def is_monotonic(ndarray[algos_t, ndim=1] arr,
+                 bint timelike) -> Tuple[bool, bool, bool]:
```

Comment:
so far at least we're not using `typing.Foo` for annotating cython code.  it might be worth doing, but its a can of worms i dont especially want to open right now

Reply:
I would be OK with this - @jbrockmendel what concerns do you have?

Reply:
cython's support for pure-python annotations, and especially mix-and-match annotations (note the arguments here are using cython-style) is fragile (https://github.com/cython/cython/issues/2529).

- I'm pretty sure cython doesnt do any optimization with this type of annotation, and we'd need to check (via asv, which is a PITA for cython functions) that we dont take an unexpected hit.
  - will readers look at this and get the incorrect impression that cython is optimizing using this annotation?
   - can we get an actually-optimizing annotation using cython-style return type `cpdef (bint, bint, bint) is_monotonic(...`?
- generated cython is super-verbose, so id want to check what this does to the size and build time of generated code
- @simonjayhawkins has looked at using stub files for mypy to use on the cython files, which I expect is the right approach.

So my opinion isn't actively "don't do this" but "this is more trouble than its worth for the time being"

Reply:
>       * can we get an actually-optimizing annotation using cython-style return type `cpdef (bint, bint, bint) is_monotonic(...`?

From the [cython docs - Typing functions](http://docs.cython.org/en/latest/src/quickstart/cythonize.html#typing-functions) it says on the bottom that typing functions can get 150 times faster over pure python (I don't know about the number if we compare "normal" cython vs typed-cython)

Reply:
> * @simonjayhawkins has looked at using stub files for mypy to use on the cython files, which I expect is the right approach.

manually creating them was rejected on grounds of maintainability, see https://github.com/pandas-dev/pandas/pull/28195#issuecomment-526623650

see https://github.com/cython/cython/issues/3150 for status of cython and stubgen

Reply:
I see, will remove the ```-> Tuple[bool, bool, bool]```